sdk/core/core-http/src/serviceClient.ts

```diff
@@ -531,6 +534,10 @@ export function serializeRequestBody(
               rootName: xmlName || serializedName
             });
           }
+        } else if (typeName === "String" && operationSpec.contentType?.match("text/plain")) {
```

Comment:
I *think* this will work:

```suggestion
        } else if (typeName === MapperType.String && operationSpec.contentType?.match("text/plain")) {
```

Reply:
Pff enums.